pf/internal/schemashim/attr_schema.go

```diff
@@ -86,39 +84,25 @@ func (*attrSchema) StateFunc() shim.SchemaStateFunc {
 
 // Needs to return a shim.Schema, a shim.Resource, or nil.
 func (s *attrSchema) Elem() interface{} {
-	t := s.attr.GetType()
-
+	switch t := s.attr.GetType().(type) {
 	// The ObjectType can be triggered through tfsdk.SingleNestedAttributes. Logically it defines an attribute with
 	// a type that is an Object type. To encode the schema of the Object type in a way the shim layer understands,
 	// Elem() needes to return a Resource value.
 	//
 	// See also: documentation on shim.Schema.Elem().
-	if tt, ok := t.(basetypes.ObjectTypable); ok {
-		var res shim.Resource = newObjectPseudoResource(tt, s.attr.Nested(), nil)
+	case basetypes.ObjectTypable:
+		var res shim.Resource = newObjectPseudoResource(t, s.attr.Nested(), nil)
 		return res
-	}
-	if tt, ok := t.(pfattr.TypeWithElementTypes); ok {
-		var res shim.Resource = newTuplePseudoResource(tt)
+	case pfattr.TypeWithElementTypes:
+		var res shim.Resource = newTuplePseudoResource(t)
 		return res
-	}
+	case pfattr.TypeWithElementType:
+		return shim.Schema(newTypeSchema(t.ElementType(), s.attr.Nested()))
 
-	// Anything else that does not have an ElementType can be skipped.
-	if _, ok := t.(pfattr.TypeWithElementType); !ok {
-		return nil
-	}
-
-	var schema shim.Schema
-	switch tt := t.(type) {
-	case types.MapType:
-		schema = newTypeSchema(tt.ElemType, s.attr.Nested())
-	case types.ListType:
-		schema = newTypeSchema(tt.ElemType, s.attr.Nested())
-	case types.SetType:
-		schema = newTypeSchema(tt.ElemType, s.attr.Nested())
+	// t does not support any kind of element type.
 	default:
-		panic(fmt.Errorf("This Elem() case is not yet supported: %v", t))
+		return nil
```

Comment:
This panic was very useful. In particular we got alerted about the missing case right away during tfgen and pointed to the right code to fix, instead of failing somewhere down the line with an obscure error. Are you sure we want to remove it?

Reply:
Previously, we would assert when:
1. `t` didn't implement `basetypes.ObjectTypable`.
2. `t` didn't implement `pfattr.TypeWithElementTypes`.
3. `t` did implement `pfattr.TypeWithElementType`
4. but was not a `types.MapType`, `types.ListType` or `types.SetType`.

 We now accept all types that implement `pfattr.TypeWithElementType`. What would the new assert criteria be?

Reply:
Ah never mind, I was a bit hasty. I like to have the switches total. But it looks like it is total already. For scalars etc the right answer from Elem() is nil it looks like.